### DISCUSSION TITLE
blog/ghost-on-fly-io-for-free-en

### DISCUSSION THREAD

**Discussion by giscus** [Pathsis/2750-words#14] [Announcements]
# blog/ghost-on-fly-io-for-free-en

This article introduce how to run a free ghost blog on fly.io to backup and restore data. Backing up and restoring data is informative for use cases on other servers.

https://rercel.com/blog/ghost-on-fly-io-for-free-en

**Comment by tribet84** [comment]
Nice, thanks for sharing! Would you mind adding MailGun support?

**Comment by Kevin-free** [comment]
感谢分享！写的很有条理。但是我还是有点问题：

1. 之前部署在其他 Ubuntu 上的 ghost，已经备份`content`到 GitHub 上，fly pull GitHub 的`content`到`/var/lib/ghost/content`，执行了`fly apps restart`重启，但是 ghost 并没有生效。另外`fly apps restart`会重置 VM 环境，每次都需要重装 git，设置密钥，有没有便捷的重启 ghost 方式？

2. 我看`fly.toml`中有

```
[mounts]
  destination = "/var/lib/ghost/content"
  source = "data"
```

这段代码的意思是将本地`data`目录拷贝到 fly 虚拟机`/var/lib/ghost/content`中对吧？但是我将`content`目录下的文件拷贝至`fly.toml`同级新建的`data`目录下，fly 虚拟机`/var/lib/ghost/content`中并没有同步。

3. 另外想请教，您在记录内容，发布到 ghost，也只能在 ghost 后台手动操作吗？能不能将内容存储在 GitHub 仓库，本地将内容 push 时自动同步到 ghost？

**Reply by Pathsis** [reply]
感谢反馈。
1. 我在[另一篇文章](https://rercel.com/blog/package-git-into-ghost-docker-image)中试图解决重启 fly app 导致 git 环境丢失的问题，不知道您是否看到？
2. 通过git备份到全新安装的ghost，重启后没有生效的原因，应该考虑两个，1）是否完全覆盖数据？2）是否修复权限（我在文章中详细提到了权限修复问题）。我建议将内容和媒体分别备份，内容（包括配置）在ghost上通过导出json来完成，媒体通过git来备份，这样只需要备份content文件夹就行了。data文件夹涉及数据库文件，可能会产生问题，不建议直接备份data文件夹。
3. 对于您的第2个问题，我以前发现，不能在安装ghost时，就将备份的数据预写入ghost。（当然，理论上是可以定制的。比如ghost默认就自带了好几篇文章。）
4. 对于您的第3个问题，我目前没有用ghost，不过有给朋友弄。现在是在ghost后台操作的，这样挺好的，否则干脆就用静态博客好啦。据我所知，有人有这样的尝试，但并不成熟也不主流。ghost依赖数据库，这决定了纯文本发布内容是很难的。

**Reply by Kevin-free** [reply]
感谢回复。
1. 刚刚看了另一篇文章，大致就是将官方镜像替换自己构建对吧。
2. 我是将`content`目录拷贝，有`var/lib/ghost/content/`权限，`data/db`文件不能备份吗？我以为是没有备份数据库文件的原因呢。根目录下的其他文件也要拷贝吗？ghost怎么导出json？管理后台没有找到，是根目录下的`config.production.json`吗？
3. OK，不能先预置，那就后备份。反正就是希望能够部署新的ghost，能将ghost设置和博客内容设为备份版。
4. 文件多，ghost后台操作稍麻烦。我找到了一个[方案](https://blog.nigzu.com/ghost-storage-editor/)：使用Obsidian+[插件](jaynguyens/obsidian-ghost-publish)，目前实测成功，虽然不是很完美，但更高效。

**Reply by Kevin-free** [reply]
还有一个题外话，您这个格式的 Table of Contents 是怎么生成的？感觉比`[TOC]`更好。应该也有快捷键吧？我在 Vscode 中编辑。

**Reply by Pathsis** [reply]
这个 Table of Contents  是网站前端实现，与编辑器没有关系。[参考](https://github.com/Lonor/site/issues/134)。

**Reply by Kevin-free** [reply]
找到了 VSCode 中有个插件[markdown-all-in-one](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one)可以实现快捷生成这样的目录
